### PR TITLE
Fix reported module name of tabular model classes

### PR DESCRIPTION
**Pull request by eslavich**
### Description

Similar to #10694, the `__module__` attribute reports "abc" for `tabular.Tabular1D` and `tabular.Tabular2D`.  This PR sets the value to "astropy.modeling.tabular" instead.

### CODE REVIEW

File: astropy/modeling/tests/test_models.py [hunk 1]
```diff
@@ -752,6 +752,16 @@ def test_tabular1d_inverse():
     assert_allclose(t(result), 100)
 
 
+@pytest.mark.skipif("not HAS_SCIPY_14")
```
Comment (pllim) [inline astropy/modeling/tests/test_models.py:755]: `HAS_SCIPY_14` can simply be `HAS_SCIPY` now. Somehow missed this one in #10345 

https://github.com/astropy/astropy/blob/ee8de6ba65e2d265dcb1d6ee48d658f3564fc99e/setup.cfg#L61

Comment (eslavich) [reply astropy/modeling/tests/test_models.py:755]: Ok, thanks, I'll update all of these in this file

**Comment by eslavich** [comment]
@pllim can we milestone this for 4.0.2?  The new asdf-astropy package won't work with versions of astropy that don't have this change.

**Review by nden — COMMENTED** [review]
Looks good to me, not sure why readthedocs is failing, may need to push a commit to restart it.

**Comment by pllim** [comment]
RTD is unrelated, see #10711 . If you are worried about doc build failing because of this PR, then for now, you have to attempt to build the docs locally.

**Comment by pllim** [comment]
RTD just fixed. I am going to close/open to kick it off again.

**Review by nden — APPROVED** [review]

**Pull request merged**